test/extensions/common/aws/credentials_provider_impl_test.cc

```diff
@@ -62,35 +62,72 @@ TEST_F(EvironmentCredentialsProviderTest, NoSessionToken) {
   EXPECT_FALSE(credentials.sessionToken().has_value());
 }
 
+class MessageMatcher : public testing::MatcherInterface<Http::RequestMessage&> {
+public:
+  explicit MessageMatcher(const Http::TestRequestHeaderMapImpl& expected_headers)
+      : expected_headers_(expected_headers) {}
+
+  bool MatchAndExplain(Http::RequestMessage& message,
+                       testing::MatchResultListener* result_listener) const override {
+    const bool equal = TestUtility::headerMapEqualIgnoreOrder(message.headers(), expected_headers_);
+    if (!equal) {
+      *result_listener << "\n"
+                       << TestUtility::addLeftAndRightPadding("Expected header map:") << "\n"
+                       << expected_headers_
+                       << TestUtility::addLeftAndRightPadding("is not equal to actual header map:")
+                       << "\n"
+                       << message.headers()
+                       << TestUtility::addLeftAndRightPadding("") // line full of padding
+                       << "\n";
+    }
+    return equal;
+  }
+
+  void DescribeTo(::std::ostream* os) const override { *os << "Message matches"; }
+
+  void DescribeNegationTo(::std::ostream* os) const override { *os << "Message does not match"; }
+
+private:
+  const Http::TestRequestHeaderMapImpl expected_headers_;
+};
+
+testing::Matcher<Http::RequestMessage&>
+MessageMatches(const Http::TestRequestHeaderMapImpl& expected_headers) {
+  return testing::MakeMatcher(new MessageMatcher(expected_headers));
+}
+
 class InstanceProfileCredentialsProviderTest : public testing::Test {
 public:
   InstanceProfileCredentialsProviderTest()
       : api_(Api::createApiForTest(time_system_)),
-        provider_(*api_,
-                  [this](const std::string& host, const std::string& path,
-                         const std::string& auth_token) -> absl::optional<std::string> {
-                    return this->fetcher_.fetch(host, path, auth_token);
-                  }) {}
+        provider_(*api_, [this](Http::RequestMessage& message) -> absl::optional<std::string> {
+          return this->fetcher_.fetch(message);
```

Comment:
Could we rename `fetcher_` to be a bit more descriptive?  Assuming this is the metadata fetcher.

Reply:
ack